File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public int[] solution(int[] arr) {
        Stack<Integer> stk = new Stack<>();
        for(int i = 0; i < arr.length;){
            if(stk.isEmpty()){
                stk.push(arr[i]); i++;
            } else if(stk.peek() == arr[i]){
                stk.pop(); i++;
            } else if(stk.peek() != arr[i]){
                stk.push(arr[i]); i++;
            }
        }
        if(stk.isEmpty()) return new int[]{-1};
        return stk.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    stk = []
    for num in arr:
        if not stk:
            stk.append(num)
        else:
            if stk[-1] == num:
                stk.pop()
            else:
                stk.append(num)
    return stk if stk else [-1]
    
# 빈배열 => arr[i]를 stk 추가하고 i++
# 마지막원소 == arr[i] => stk pop, i++
# 마지막원소 != arr[i] => stk push arr[i], i++